Source/FullScreenPokemon.ts

```diff
@@ -9196,6 +9196,41 @@ module FullScreenPokemon {
         }
 
         /**
+         * Pushes and saves the current state of a variable to a stack.
+         * 
+         * @param thing   The Thing, Area, Map, or Location saving its state of a variable.
+         * @param title   Name for the state being saved.
+         * @param value   The values of the variable to be saved.
+         */
+        addStateHistory(thing: IStateSaveable, title: string, value: any): void {
+            if (!thing.state) {
+                thing.state = {};
+            }
+
+            let stateHistory: any[] = thing.state[title];
+            if (stateHistory) {
+                stateHistory.push(value);
+            } else {
+                thing.state[title] = [value];
+            }
+        }
+
+        /**
+         * Updates to the most recently saved state for a variable.
+         * 
+         * @param thing   The Thing having its state restored.
+         * @param title   The name of the state to restore.
+         */
+        popStateHistory(thing: IStateSaveable, title: string): void {
+            let stateHistory: any[] = thing.state[title];
+            if (stateHistory && stateHistory.length > 0) {
+                thing[title] = stateHistory.pop();
+            } else {
+                throw new Error("No state saved for " + title + ".");
```

Comment:
Nit: inverse the `if` statement so this throw happens first. That way there's no need for an `else`. Slightly cleaner